.github/labeler.yml

```diff
@@ -170,3 +173,104 @@ module:sql:
   - changed-files:
       - any-glob-to-any-file:
           - solr/modules/sql/**
+
+# Any test source files
+tests:
+  - changed-files:
+      - any-glob-to-any-file:
+          - '**/src/test/**'
+          - '**/src/test-files/**'
+
+# Add category labels loosely based on java packages
+cat:analysis:
+  - changed-files:
+      - any-glob-to-any-file:
+          - '**/org/apache/solr/analysis/**'
+
+cat:cli:
+  - changed-files:
+      - any-glob-to-any-file:
+          - '**/org/apache/solr/cli/**'
+
+cat:cloud:
+  - changed-files:
+      - any-glob-to-any-file:
+          - '**/org/apache/solr/cloud/**'
+          - '**/org/apache/solr/cluster/**'
+
+cat:backup:
+  - changed-files:
+      - any-glob-to-any-file:
+          - '**/org/apache/solr/core/backup/**'
+          - '**/org/apache/solr/hdfs/backup/**'
+
+cat:core:
+  - changed-files:
+      - any-glob-to-any-file:
+          - '**/org/apache/solr/core/**'
+
+cat:filestore:
+  - changed-files:
+      - any-glob-to-any-file:
+          - '**/org/apache/solr/filestore/**'
+
+cat:handler:
+  - changed-files:
+      - any-glob-to-any-file:
+          - '**/org/apache/solr/handler/**'
+
+cat:index:
+  - changed-files:
+      - any-glob-to-any-file:
+          - '**/org/apache/solr/index/**'
+          - '**/org/apache/solr/hdfs/index/**'
+
+cat:logging:
+  - changed-files:
+      - any-glob-to-any-file:
+          - '**/org/apache/solr/logging/**'
+
+cat:metrics:
+  - changed-files:
+      - any-glob-to-any-file:
+          - '**/org/apache/solr/metrics/**'
+
+cat:packagemanager:
+  - changed-files:
+      - any-glob-to-any-file:
+          - '**/org/apache/solr/packagemanager/**'
+          - '**/org/apache/solr/pkg/**'
+
+cat:schema:
+  - changed-files:
+      - any-glob-to-any-file:
+          - '**/org/apache/solr/rest/**'
+          - '**/org/apache/solr/schema/**'
+
+cat:search:
+  - changed-files:
+      - any-glob-to-any-file:
+          - '**/org/apache/solr/search/**'
+          - '**/org/apache/solr/highlight/**'
+          - '**/org/apache/solr/request/**'
+          - '**/org/apache/solr/response/**'
+          - '**/org/apache/solr/parser/**'
+          - '**/org/apache/solr/query/**'
+          - '**/org/apache/solr/spelling/**'
+
+cat:security:
+  - changed-files:
+      - any-glob-to-any-file:
+          - '**/org/apache/solr/security/**'
+
+cat:servlet:
+  - changed-files:
+      - any-glob-to-any-file:
+          - '**/org/apache/solr/servlet/**'
+
+cat:update:
```

Comment:
Perhaps `index` can go into `update`. The `index` package is mostly lucene merge policies...